docs/source/appendix.rst

```diff
@@ -8,18 +8,102 @@ Glossary
 .. glossary::
     :sorted:
 
-    `ADES`: |ades|
-    `AWS`: Amazon Web Services
-    `CWL`_: |cwl|
-    `EMS`: |ems|
-    `ESGF`_: |esgf|
-    `ESGF-CWT`_: |esgf-cwt|
-    `HREF`: Hyperlink Reference
-    I/O: Inputs and/or Outputs of CWL and/or WPS depending on context
-    `OGC`_: |ogc|
-    `S3`: Simple Storage Service (AWS S3), bucket file storage
-    `WKT`: Well-Known Text geometry representation
-    `WPS`: Web Processing Service
+    ADES
+        | |ades|
+        | See :ref:`processes` section for details, as well as :term:`EMS` for alternative operation mode.
+
+    Application Package
+        General term that refers to *"what and how the :term:`Process` will execute"*. Application Packages provide
+        the core details about the execution methodology of the underlying operation the :term:`Process` provides, and
+        are therefore always contained within a :term:`Process` definition. This is more specifically represented
+        by a :term:`CWL` specification in the case of `Weaver` implementation, but could technically be defined by
+        another similar approach. See :ref:`Application Package` section for all relevant details.
+
+    AWS
+        Amazon Web Services
+
+    CWL
+        | |cwl|_
+        | Representation of the internal :term:`Application Package` of the :term:`Process` to provide execution
+          methodology of the referenced :term:`Docker` image or other supported definitions. See
+          :ref:`application-package` section for further details.
+
+    Docker
+        Image that packages all required dependencies of an application or software in order to correctly execute it
+        in a containerized and isolated virtualized environment.
```

Comment:
Since it's not an image itself, I would reformulate the phrase. 
Something like "Containerized and isolated environment platform that allows all required dependencies of an application or software to be packaged in a single image in order to correctly execute the virtualized application."

Reply:
I like that. +1